src/libaktualizr/uptane/tuf.cc

```diff
@@ -209,6 +215,61 @@ bool Target::IsOstree() const {
   }
 }
 
+bool Target::operator==(const Target &t2) const {
```

Comment:
At first glance, it's confusing that a single class represents two types of Targets that require different context/class members, as well as their comparison.

Reply:
That is an interesting point. @lbonn what would you think of having two children classes, one for DirectorTarget and one for ImagesTarget? Almost all of the logic would still be in the parent class, but the ecu map and hwid vectors would be in the respective children, and the match function would probably have to be a static function outside of the classes. I think that would help make things much, much more explicit.

Reply:
On second thought, this is a pretty huge change and will require changing the API. I'm deep enough in this task already that I think I'd rather wait on that work and come back to it another time.